feat: add pagination to statement segments list for performance

mussbach opened the pull request:
## Summary
* Add DpPager component with page size options (10, 20, 50 segments per page)
* Implement pagination mixin for consistent state management with other lists
* Add pagination parameters to segment API calls to limit data loading
* Include page change and size change handlers for smooth navigation
* Store pagination state in localStorage per statement for user preference persistence

## Performance Impact
Resolves performance issues when viewing statements with 5000+ segments:
- **Before**: Loads ALL segments at once causing long delays and memory issues
- **After**: Loads only 20 segments per page by default with user-selectable page sizes

## Test plan
- [ ] Verify segments list loads quickly with large segment counts (5000+)
- [ ] Test pagination controls work correctly (page navigation, size changes)
- [ ] Confirm pagination state persists across page reloads
- [ ] Validate all segments are accessible through pagination
- [ ] Test that URL segment parameters still work for deep linking

hwiem reviewed (commented):
The code looks good, but I'd prefer to move the pagination into the sticky header, that's how we solved it in similar views (statement list, segments list) - when the user scrolls down, they still see the pagination; with the current implementation, they might not realize that there are more segments on other pages.
I don't know if there's time for that, because it would also require moving the `fetchSegments` call to StatementSegmentsList.vue - but that might also improve performance, since at the moment we fetch the segments both in SegmentsRecommendations.vue and StatementSegmentsEdit.vue, i.e. in both the "Erwiderung" and "Bearbeiten" tabs.
If that's too much, I'd consider duplicating the pager below the segments list so that it's always visible and hiding it entirely when there is only one page. And I think it should be displayed above the "Abschnitt" / "Erwiderung" table header. What do you think?

mussbach commented:
@hwiem thank you for the review. I agree that the suggested change would be best. I checked it and the change would be quite huge, so i first implemented your alternative solution in [4190b73](https://github.com/demos-europe/demosplan-core/pull/5268/commits/4190b7399135369ed91ad1a471c2cecba1cf65a8)

hwiem reviewed (commented):
I just noticed two issues: When I navigate to the statement segments list, I first see the paginated list of segments in the "Erwiderung" tab. I toggle to the "Bearbeiten" tab - there are only 10 segments, no pagination (there should be 20 segments). I navigate back to the "Erwiderung" tab, and there are only 10 segments and no pagination. Maybe adding pagination to the "Bearbeiten" tab could already fix it?
(The issue that the pagination disappears is also resolved if you remove the
```
if (Object.keys(this.segments).length === 0)
```
on mounted (`SegmentsRecommendations.vue`) and just call `fetchSegments` always, but that of course makes toggling between tabs slower)

mussbach commented:
## Response to Conditional Fetching Suggestion

Regarding the suggestion to add conditional fetching:

```javascript
if (Object.keys(this.segments).length === 0) {
  this.fetchSegments(this.pagination?.currentPage || 1)
}
```

I've decided **not** to implement this optimization because:

### Problem with the Check

The `this.segments` variable maps to the Vuex store's `StatementSegment/items`, which contains **all segments across all statements** that have been loaded during the session, not just segments for the current statement.

### Resulting Bug

If you navigate from Statement A (with segments) → Statement B (different segments):
1. ✅ Statement A loads correctly (`segments` is empty initially)
2. ❌ Statement B would **not fetch** because `segments` is no longer empty
3. ❌ Statement B would incorrectly display Statement A's segments

### Current Behavior is Correct

The current implementation:
- Always fetches segments filtered by `parentStatement.id` (lines 309-320 in SegmentsRecommendations.vue)
- The backend filtering ensures only the correct statement's segments are returned
- Any small overhead from re-fetching is negligible compared to correctness

### Note
This matches your comment that it was "not a blocker" and "didn't test this" - the suggestion would indeed require changes to how segments are stored/filtered in Vuex to work correctly.

---

✅ I've implemented the other two suggestions:
1. Removed `pagination.totalPages > 1` check (fixes the visibility bug)
2. Added `await` to `nextTick` calls (improves robustness)

mussbach commented on client/js/components/procedure/StatementSegmentsList/StatementSegmentsEdit.vue:
```diff
@@ -456,15 +500,40 @@ export default {
+
+      this.isLoading = false
+
+      this.$nextTick(() => {
```
[3a9e41b](https://github.com/demos-europe/demosplan-core/pull/5268/commits/3a9e41bdfa19b636d000899e900362129f309a48)

mussbach commented on client/js/components/procedure/StatementSegmentsList/StatementSegmentsEdit.vue:
```diff
@@ -456,15 +500,40 @@ export default {
+  mounted () {
+    if (hasPermission('area_statement_segmentation')) {
+      this.initPagination()
+      this.fetchSegments(this.pagination?.currentPage || 1)
```
see comment below

mussbach commented on client/js/components/procedure/StatementSegmentsList/SegmentsRecommendations.vue:
```diff
@@ -11,6 +11,22 @@
     <div v-else>
+      <!-- Pagination above table header -->
+      <div
+        v-if="pagination && pagination.currentPage && pagination.totalPages > 1"
```
[3a9e41b](https://github.com/demos-europe/demosplan-core/pull/5268/commits/3a9e41bdfa19b636d000899e900362129f309a48)

hwiem reviewed (approved):
Nice ✨ Thank you!
